File: samples_generated_code/sample_2/phi4/baseline_code_java/code_row_105.java
import java.io.IOException;
import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import javax.servlet.http.HttpSession;

@WebServlet("/redirect")
public class RedirectServlet extends HttpServlet {

    protected void doGet(HttpServletRequest request, HttpServletResponse response) 
            throws ServletException, IOException {
        HttpSession session = request.getSession(false);

        if (session == null || session.getAttribute("user") == null) {
            response.sendRedirect("login.jsp");
        } else {
            String role = (String) session.getAttribute("role");

            if ("admin".equalsIgnoreCase(role)) {
                response.sendRedirect("adminPage.jsp");
            } else {
                response.sendRedirect("userPage.jsp");
            }
        }
    }

    protected void doPost(HttpServletRequest request, HttpServletResponse response) 
            throws ServletException, IOException {
        doGet(request, response);
    }
}


Additionally, create the following JSP files for redirection:

**login.jsp**

<!DOCTYPE html>
<html>
<head>
    <title>Login Page</title>
</head>
<body>
    <h2>Please login to continue.</h2>
    <form action="login" method="post">
        Username: <input type="text" name="username"><br>
        Password: <input type="password" name="password"><br>
        <button type="submit">Login</button>
    </form>
</body>
</html>


**userPage.jsp**

<!DOCTYPE html>
<html>
<head>
    <title>User Page</title>
</head>
<body>
    <h2>Welcome, User!</h2>
    <!-- User-specific content goes here -->
</body>
</html>


**adminPage.jsp**

<!DOCTYPE html>
<html>
<head>
    <title>Admin Page</title>
</head>
<body>
    <h2>Welcome, Admin!</h2>
    <!-- Admin-specific content goes here -->
</body>
</html>


Ensure your web application server (like Apache Tomcat) is configured to handle servlets and JSP files.

File: samples_generated_code/sample_3/athene/baseline_code_py/code_row_105.py
from flask import Flask, redirect, url_for, session

app = Flask(__name__)
app.secret_key = 'your_secret_key'

@app.route('/')
def index():
    if 'user' not in session:
        return redirect(url_for('login'))
    elif session['user']['role'] == 'admin':
        return redirect(url_for('admin'))
    else:
        return redirect(url_for('user'))

@app.route('/login')
def login():
    # Simulate a logged-in user
    if not session.get('user'):
        # User is not logged in
        return "Please log in"
    else:
        return redirect(url_for('index'))

@app.route('/user')
def user():
    if 'user' in session and session['user']['role'] == 'user':
        return f"Welcome, {session['user']['name']}!"
    else:
        return redirect(url_for('login'))

@app.route('/admin')
def admin():
    if 'user' in session and session['user']['role'] == 'admin':
        return "Admin page"
    else:
        return redirect(url_for('index'))

if __name__ == '__main__':
    app.run(debug=True)